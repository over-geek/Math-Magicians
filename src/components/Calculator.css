.container {
  width: 400px;
  height: 500px;
  margin: 50px auto;
  border: 1px solid #ddd;
}

.calc-result {
  height: 10%;
  display: flex;
  justify-content: flex-end;
  overflow: auto;
  background-color: rgba(0, 0, 0, 0.488);
  color: #fff;
  padding: 10px 0;
  padding-right: 5px;
  text-align: right;
}

.btn-17 {
  grid-column: 1 / 3;
}

.btn-4,
.btn-8,
.btn-12,
.btn-16,
.btn-19 {
  background-color: #f5913e;
}

.buttons-container {
  height: 90%;
  display: flex;
  flex-direction: column;
}

.button {
  flex: 1;
  display: grid;
  grid-template-columns: repeat(4, 1fr);
}

button {
  border: 1px solid #ddd;
  cursor: pointer;
  outline: none;
  padding: 10px 0;
}
