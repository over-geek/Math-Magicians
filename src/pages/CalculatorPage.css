.calc-page-container {
  margin-top: 2rem;
  display: flex;
  justify-content: space-between;
}