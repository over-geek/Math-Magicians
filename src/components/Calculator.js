import { useState } from 'react';
import './styles/Calculator.css';
import Button from './Button';
import {
  OPERATOR,
  NUM_1,
  NUM_2,
  NUM_3,
  NUM_4,
} from '../utils/constants';
import calculate from '../logic/calculate';

function Calculator() {
  const initialState = {
    operator: null,
    next: null,
    total: null,
  };

  const [dataObj, setDataObj] = useState(initialState);
  const operations = (text) => {
    const result = calculate(dataObj, text);
    setDataObj(result);
  };

  return (
    <div className="calc-page-container">
      <div>
        <p>Let&apos;s do some Math!</p>
      </div>
      <div className="calc-container">
        <div className="calc-result">{`${dataObj.next || dataObj.total || 0}`}</div>
        <div className="buttons-container">
          <div className="button">
            {OPERATOR.map((item) => (<Button key={item.id} title={item.op} className={`btn-${item.id}`} interactivity={operations} />))}
          </div>
          <div className="button">
            {NUM_1.map((item) => (<Button key={item.id} title={item.num} className={`btn-${item.id}`} interactivity={operations} />))}
          </div>
          <div className="button">
            {NUM_2.map((item) => (<Button key={item.id} title={item.num} className={`btn-${item.id}`} interactivity={operations} />))}
          </div>
          <div className="button">
            {NUM_3.map((item) => (<Button key={item.id} title={item.num} className={`btn-${item.id}`} interactivity={operations} />))}
          </div>
          <div className="button">
            {NUM_4.map((item) => (<Button key={item.id} title={item.num} className={`btn-${item.id}`} interactivity={operations} />))}
          </div>
        </div>
      </div>
    </div>
  );
}

export default Calculator;
