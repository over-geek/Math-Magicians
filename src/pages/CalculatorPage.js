import './CalculatorPage.css';
import Calculator from '../components/Calculator';

const CalculatorPage = () => (
  <div className="calc-page-container">
    <h2>Let&apos;s do some Math!</h2>
    <Calculator />
  </div>
);

export default CalculatorPage;
