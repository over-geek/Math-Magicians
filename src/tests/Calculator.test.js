import React from 'react';
import '@testing-library/jest-dom/extend-expect';
import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';
import Calculator from '../components/Calculator';
import CalculatorPage from '../pages/CalculatorPage';

describe('Calculator component', () => {
  test('renders correctly into DOM', () => {
    const component = renderer.create(
      <MemoryRouter>
        <Calculator />
      </MemoryRouter>,
    ).toJSON();
    expect(component).toMatchSnapshot();
  });

  test('renders the text, \'Let\'s do some Math!\'', () => {
    const component = render(
      <MemoryRouter>
        <CalculatorPage />
      </MemoryRouter>,
    );

    expect(component.container).toHaveTextContent('Let\'s do some Math!0AC+/-%รท789x456-123+0.=');
  });
});
