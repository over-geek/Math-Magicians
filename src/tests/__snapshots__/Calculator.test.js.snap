// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calculator component renders correctly into DOM 1`] = `
<div
  className="calc-container"
>
  <div
    className="calc-result"
  >
    0
  </div>
  <div
    className="buttons-container"
  >
    <div
      className="button"
    >
      <button
        className="btn-1"
        onClick={[Function]}
        type="button"
      >
        AC
      </button>
      <button
        className="btn-2"
        onClick={[Function]}
        type="button"
      >
        +/-
      </button>
      <button
        className="btn-3"
        onClick={[Function]}
        type="button"
      >
        %
      </button>
      <button
        className="btn-4"
        onClick={[Function]}
        type="button"
      >
        รท
      </button>
    </div>
    <div
      className="button"
    >
      <button
        className="btn-5"
        onClick={[Function]}
        type="button"
      >
        7
      </button>
      <button
        className="btn-6"
        onClick={[Function]}
        type="button"
      >
        8
      </button>
      <button
        className="btn-7"
        onClick={[Function]}
        type="button"
      >
        9
      </button>
      <button
        className="btn-8"
        onClick={[Function]}
        type="button"
      >
        x
      </button>
    </div>
    <div
      className="button"
    >
      <button
        className="btn-9"
        onClick={[Function]}
        type="button"
      >
        4
      </button>
      <button
        className="btn-10"
        onClick={[Function]}
        type="button"
      >
        5
      </button>
      <button
        className="btn-11"
        onClick={[Function]}
        type="button"
      >
        6
      </button>
      <button
        className="btn-12"
        onClick={[Function]}
        type="button"
      >
        -
      </button>
    </div>
    <div
      className="button"
    >
      <button
        className="btn-13"
        onClick={[Function]}
        type="button"
      >
        1
      </button>
      <button
        className="btn-14"
        onClick={[Function]}
        type="button"
      >
        2
      </button>
      <button
        className="btn-15"
        onClick={[Function]}
        type="button"
      >
        3
      </button>
      <button
        className="btn-16"
        onClick={[Function]}
        type="button"
      >
        +
      </button>
    </div>
    <div
      className="button"
    >
      <button
        className="btn-17"
        onClick={[Function]}
        type="button"
      >
        0
      </button>
      <button
        className="btn-18"
        onClick={[Function]}
        type="button"
      >
        .
      </button>
      <button
        className="btn-19"
        onClick={[Function]}
        type="button"
      >
        =
      </button>
    </div>
  </div>
</div>
`;
